packages/controllers/src/snaps/SnapController.ts

```diff
@@ -249,9 +250,11 @@ export class SnapController extends BaseController<
 
   private _maxIdleTime: number;
 
+  private _maxRequestTime: number;
+
   private _idleTimeCheckInterval: number;
 
-  private _timeoutForLastRequestStatus?: NodeJS.Timeout;
+  private _timeoutForLastRequestStatus?: number;
```

Comment:
using `window.setTimeout` lets me just use `number`: https://stackoverflow.com/a/50428377

Reply:
That is nice, however we shouldn't enshrine the `window` object like this in our controllers. We may not run them in the browser in the future.